WcaOnRails/app/views/registrations/_register_form.html.erb

```diff
@@ -0,0 +1,38 @@
+<% url = @registration.new_record? ? competition_registrations_path(@competition)
+                                   : registration_path(@registration) %>
+<%= horizontal_simple_form_for @registration, url: url, html: { class: 'are-you-sure no-submit-on-enter' } do |f| %>
+  <% selected_events = @registration.saved_and_unsaved_events.empty? ? @registration.user.preferred_events : @registration.saved_and_unsaved_events %>
+  <% notify_of_preferred_events = if current_user.preferred_events.empty?
```

Comment:
I think computing `selected_events` and `notify_of_preferred_events` belong in the controller.